src/Data/CreditCard.hs

```diff
@@ -370,7 +370,9 @@ guessCreditCardType n
   | otherwise      = Nothing
   where
     isVisa         = (4 == firstDigits 1) && (digitsLength `elem` [13,16,19])
-    isMasterCard   = (inRange (51, 55) (firstDigits 2)) && (digitsLength == 16)
+    isMasterCard   =
+      (inRange (51, 55) (firstDigits 2)) && (digitsLength == 16)
+      || (inRange (2221, 2720) (firstDigits 4)) && (digitsLength == 16)
```

Comment:
Hmm... Just realized... Are parens placed correctly? Shouldn't it be
``` haskell
(inRange (51, 55) (firstDigits 2) && digitsLength == 16) || ...
```
?
Probably operators associativity does the right job, but I would not rely on that.
On a side note in complex cases I prefer smth like 
``` haskell
or 
  [ p1 && p2
  , p3 && p4
  ] 
```
exactly to not dive into parens mess

Reply:
Tried to match style of the original, but it can be rewritten np.